Sources/Datadog/RUM/Debugging/RUMDebugging.swift

```diff
@@ -0,0 +1,185 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+internal protocol RUMDebugging {
+    func debug(applicationScope: RUMApplicationScope)
+}
+
+#if targetEnvironment(simulator)
+import UIKit
+import Foundation
+
+private struct RUMDebugInfo {
+    struct View {
+        let uri: String
+        let isActive: Bool
+
+        init(scope: RUMViewScope) {
+            self.uri = scope.viewURI
+            self.isActive = scope.isActiveView
+        }
+    }
+
+    let views: [View]
+
+    init(applicationScope: RUMApplicationScope) {
+        self.views = (applicationScope.sessionScope?.viewScopes ?? [])
+            .map { View(scope: $0) }
+    }
+}
+
+internal class RUMDebuggingInSimulator: RUMDebugging {
+    private lazy var canvas: UIView = {
+        let window = UIApplication.shared.keyWindow
+        let view = RUMDebugView(frame: window?.bounds ?? .zero)
+        view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        window?.addSubview(view)
+        return view
+    }()
+
+    // MARK: - Initialization
+
+    init() {
+        UIDevice.current.beginGeneratingDeviceOrientationNotifications()
+
+        NotificationCenter.default
+            .addObserver(
+                self,
+                selector: #selector(RUMDebuggingInSimulator.updateLayout),
+                name: UIDevice.orientationDidChangeNotification,
```

Comment:
❔  is this an `UIKit` bug that this event not triggerring `layoutSubviews` calls?

Reply:
I think it gets called if the `UIView` is managed by the view controller. Here we just add the view to the `UIWindow`.